itest/rust/src/object_tests/property_test.rs

```diff
@@ -367,4 +368,127 @@ fn derive_export() {
+
+#[derive(GodotClass)]
+#[class(base=Resource)]
+pub struct CustomResource {}
+
+#[godot_api]
+impl CustomResource {}
+
+#[godot_api]
+impl ResourceVirtual for CustomResource {
+    fn init(_base: godot::obj::Base<Self::Base>) -> Self {
+        Self {}
+    }
+}
+
+#[derive(GodotClass)]
+#[class(base=Resource, rename=NewNameCustomResource)]
+pub struct RenamedCustomResource {}
+
+#[godot_api]
+impl RenamedCustomResource {}
+
+#[godot_api]
+impl ResourceVirtual for RenamedCustomResource {
+    fn init(_base: godot::obj::Base<Self::Base>) -> Self {
+        Self {}
+    }
+}
+
+#[derive(GodotClass)]
+#[class(base=Node)]
+pub struct ExportResource {
+    #[export]
+    #[var(usage_flags=[PROPERTY_USAGE_DEFAULT, PROPERTY_USAGE_EDITOR_INSTANTIATE_OBJECT])]
+    pub foo: Option<Gd<CustomResource>>,
+
+    #[export]
+    pub bar: Option<Gd<RenamedCustomResource>>,
+}
+
+#[godot_api]
+impl ExportResource {}
+
+#[godot_api]
+impl NodeVirtual for ExportResource {
+    fn init(_base: godot::obj::Base<Self::Base>) -> Self {
+        Self {
+            foo: None,
+            bar: None,
+        }
+    }
+}
+
+#[itest]
+fn export_resource() {
+    let class: Gd<ExportResource> = Gd::new_default();
+
+    let property = class
+        .get_property_list()
+        .iter_shared()
+        .find(|c| c.get_or_nil("name") == "foo".to_variant())
+        .unwrap();
+    // `class_name` should be empty for non-Object variants.
+    assert_eq!(
+        property.get_or_nil("class_name"),
+        "CustomResource".to_variant()
+    );
+    assert_eq!(
+        property.get_or_nil("type"),
+        (VariantType::Object as i32).to_variant()
+    );
+    assert_eq!(
+        property.get_or_nil("hint"),
+        (PropertyHint::PROPERTY_HINT_RESOURCE_TYPE.ord()).to_variant()
+    );
+    assert_eq!(
+        property.get_or_nil("hint_string"),
+        "CustomResource".to_variant()
+    );
+    assert_eq!(
+        property.get_or_nil("usage"),
+        (PropertyUsageFlags::PROPERTY_USAGE_DEFAULT.ord()
+            | PropertyUsageFlags::PROPERTY_USAGE_EDITOR_INSTANTIATE_OBJECT.ord())
+        .to_variant()
+    );
+
+    let property = class
+        .get_property_list()
+        .iter_shared()
+        .find(|c| c.get_or_nil("name") == "bar".to_variant())
+        .unwrap();
+    // `class_name` should be empty for non-Object variants.
+    assert_eq!(
+        property.get_or_nil("class_name"),
+        "NewNameCustomResource".to_variant()
+    );
+    assert_eq!(
+        property.get_or_nil("type"),
+        (VariantType::Object as i32).to_variant()
+    );
+    assert_eq!(
+        property.get_or_nil("hint"),
+        (PropertyHint::PROPERTY_HINT_RESOURCE_TYPE.ord()).to_variant()
+    );
+    assert_eq!(
+        property.get_or_nil("hint_string"),
+        "NewNameCustomResource".to_variant()
+    );
+    assert_eq!(
+        property.get_or_nil("usage"),
+        PropertyUsageFlags::PROPERTY_USAGE_DEFAULT
+            .ord()
+            .to_variant()
+    );
```

Comment:
You have a lot of code following the same pattern, which can be extracted into a function.
```rs
fn check_property(property: &Variant, key: &str, expected: impl ToVariant) {
    assert_eq!(property.get_or_nil(key), expected.to_variant());
}
```

This would then simply the code to:
```suggestion
    // `class_name` should be empty for non-Object variants.
    check_property(&property, "class_name", "CustomResource");
    check_property(&property, "type", VariantType::Object as i32);
    check_property(&property, "hint", PropertyHint::PROPERTY_HINT_RESOURCE_TYPE.ord());
    check_property(&property, "hint_string", "CustomResource");
    check_property(&property, "usage", PropertyUsageFlags::PROPERTY_USAGE_DEFAULT.ord()
        | PropertyUsageFlags::PROPERTY_USAGE_EDITOR_INSTANTIATE_OBJECT.ord());

    let property = class
        .get_property_list()
        .iter_shared()
        .find(|c| c.get_or_nil("name") == "bar".to_variant())
        .unwrap();
        
    // `class_name` should be empty for non-Object variants.
    check_property(&property, "class_name", "NewNameCustomResource");
    check_property(&property, "type", VariantType::Object as i32);
    check_property(&property, "hint", PropertyHint::PROPERTY_HINT_RESOURCE_TYPE.ord());
    check_property(&property, "hint_string", "NewNameCustomResource");
    check_property(&property, "usage", PropertyUsageFlags::PROPERTY_USAGE_DEFAULT.ord());
```
(Maybe check for mistakes)